blocks/library/chat/chat-transcript.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { uniq } from 'lodash';
+
+export default function ChatTranscript( props ) {
+	const { value, compact } = props;
+
+	let lastAuthor;
+	let authors = [];
+
+	function getAuthor( author ) {
+		author = author.trim();
+		authors.push( author );
+
+		authors = uniq( authors );
+
+		return authors.indexOf( author ) + 1;
+	}
+
+	return (
+		<div className={ `wp-block-chat${ compact ? ' is-compact' : '' }` }>
+			{value && value.split( '\n' ).map( ( line, index ) => {
+				line = line.trim();
+
+				let message = line;
+				let author;
+
+				if ( line.includes( ':' ) ) {
+					// Includes an author.
+					const msg = line.split( ':' );
+					author = msg.shift().trim();
+					message = msg.join( ':' ).trim();
+
+					lastAuthor = getAuthor( author );
+				}
```

Comment:
Do you think we could split the logic of this into two parts?

```js
// Normalize data
const transcript = normalizeTranscript( value ); // returns { authors: [], messages: [ { message, authorIndex } ] }

// Render
return messages.map( ... )
```

This would make unit-testing the normalizing logic easier.